src/client/datascience/prereleaseChecker.ts

```diff
@@ -20,12 +20,12 @@ export class PreReleaseChecker implements IExtensionSingleActivationService {
         @inject(IApplicationEnvironment) private readonly appEnv: IApplicationEnvironment,
         @inject(IApplicationShell) private readonly appShell: IApplicationShell,
         @inject(IMemento) @named(GLOBAL_MEMENTO) private globalState: vscode.Memento,
-        @inject(IsPreRelease) private isPreRelease: boolean
+        @inject(IsPreRelease) private isPreRelease: Promise<boolean>
     ) {}
     public async activate(): Promise<void> {
         // Ask user if the version is not prerelease
         const dontAsk = this.globalState.get(PRERELEASE_DONT_ASK_FLAG, false);
-        if (!this.isPreRelease && this.appEnv.channel === 'insiders' && !dontAsk) {
+        if (!(await this.isPreRelease) && this.appEnv.channel === 'insiders' && !dontAsk) {
```

Comment:
We probably should not await here in `activate`, as this slows loading the entire Jupyter extension.
This could happen in the background.
E.g. 
```typescript
public asycn activate() {
	void this.initialize();
}
private async initialize() {
	await this.isPreRelease
}
```

This way, the jupyter extension will activate without having to awit for the FS operation to complete.
Else its blocked & slows down, I know its not much, but death by a thousand cuts

Reply:
Sure good idea.